### ISSUE TITLE
CKV_AWS_111 - cloudformation  check cloudwatch:PutMetricData

### ISSUE THREAD

**Issue by cbuto** [bridgecrewio/checkov#990]
**Describe the bug**
When running checkov against a cloudformation template with an IAM policy including the action `cloudwatch:PutMetricData` and a condition limiting the action to a specific namepace, the `CKV_AWS_111` rule is triggered. Since [PutMetricData](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncloudwatch.html#amazoncloudwatch-PutMetricData) doesn't support limiting by a resource, the condition seems like it should be a sufficient control to limit write access. 

**To Reproduce**
1. Create a Cloudformation template containing a policy that looks like:
```
- Sid: AllowPutMetricData
  Effect: Allow
  Action:
    - cloudwatch:PutMetricData
  Resource: "*"
  Condition:
    StringEquals:
      cloudwatch:namespace: Namespace
```
2. Run `checkov -d </path/to/cloudformation> --quiet`
3. You should see checkov flag the policy as a violation of `CKV_AWS_111`

**Expected behavior**
I expect checkov to consider that policy valid for the `CKV_AWS_111` rule.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Checkov Version 1.0.844

Thanks ahead of time for reviewing this and thanks for the awesome tool!

**Comment by schosterbarak** [comment]
@kmcquade should `policy.write_actions_without_constraints` take into account `Condition`s if `Resource: "*"` since the condition is acting as a constraint? 

or is it already doing it and I have a bug? :S

**Comment by kmcquade** [comment]
@schosterbarak - Cloudsplaining doesn't filter based on conditions, actually. It was a trade off made at design time. I'm open to suggestions on how we might handle this, though.

**Comment by yi2020** [comment]
@kmcquade our approach with Cloudrail has been to assume that if there are conditions, it's not really *. We do in some cases dig through the condition to see if it really limits anything, but the amount of logic required was insane.

**Comment by schosterbarak** [comment]
I'm up for following @yi2020 approach.

@kmcquade I think it's better to assume that if a condition is set - the check should pass. I can contribute that logic to cloudsplaining if you would like or implemented it just in checkov. Wdyt?

**Comment by kmcquade** [comment]
@schosterbarak I like that idea from @yi2020 too. If you want to contribute that to Cloudsplaining, we can fix it in both places!

**Comment by schosterbarak** [comment]
@kmcquade @yi2020 @cbuto PR created: https://github.com/salesforce/cloudsplaining/pull/167
